### PR TITLE
feat: lists

### PR DESCRIPTION
**Pull request by yggg**
### Please read and mark the following check list before creating a pull request:

 - [x] I read and followed the [CONTRIBUTING.md](https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md) guide.
 - [x] I read and followed the [New Feature Checklist](https://github.com/akveo/nebular/blob/master/DEV_DOCS.md#new-feature-checklist) guide.
 
 #### Short description of what this resolves:
List and infinite list components

### CODE REVIEW

File: src/framework/theme/components/list/list.component.ts [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/**
+ * List is a container component that wraps `nb-list-item` or `nb-list-items-group` components.
+ *
+ * Basic example:
+ * @stacked-example(Simple list, list/simple-list-showcase.component)
+ *
+ * `nb-list-item` accepts arbitrary content, so you can create list of any components.
+ *
+ * List of users:
+ * @stacked-example(Users list, list/users-list-showcase.component)
```
Comment (nnixaa) [inline src/framework/theme/components/list/list.component.ts]: `@theme` block is missed

File: src/framework/theme/components/card/card.component.ts [hunk 2]
```diff
@@ -73,6 +73,13 @@ export class NbCardFooterComponent {
  * Card with header and footer:
  * @stacked-example(With Header & Footer, card/card-full.component)
  *
+ * Most of the time main card content goes to `nb-card-body`, so it styled and aligned according to header and footer.
```
Comment (nnixaa) [inline src/framework/theme/components/card/card.component.ts]: so it _is_ styled and aligned according_ly_ to header and footer

File: src/framework/theme/components/card/card.component.ts [hunk 3]
```diff
@@ -73,6 +73,13 @@ export class NbCardFooterComponent {
...
+ * But in case you need a higher level of control, you can pass contend directly to `nb-card`,
```
Comment (nnixaa) [inline src/framework/theme/components/card/card.component.ts]: ~But~ In

File: src/framework/theme/components/list/list.component.ts (hunk 1 above)
Comment (yggg) [reply src/framework/theme/components/list/list.component.ts]: done

File: src/framework/theme/components/card/card.component.ts [hunk 4]
```diff
@@ -73,6 +73,13 @@ export class NbCardFooterComponent {
  * Card with header and footer:
  * @stacked-example(With Header & Footer, card/card-full.component)
  *
+ * Most of the time main card content goes to `nb-card-body`, so it styled and aligned accordingly to header and footer.
```
Comment (nnixaa) [inline src/framework/theme/components/card/card.component.ts]: so it is styled and aligned in accordance with the header and footer.

File: src/framework/theme/components/list/infinite-list.directive.ts [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+      this.topThreshold.emit();
+    }
+
+    this.lastScrollPosition = scrollTop;
+  }
+
+  private getContainerDimentions(): Observable<NbScrollableContainerDimentions> {
+    if (this.elementScroll) {
+      const { scrollTop, scrollHeight, clientHeight } = this.elementRef.nativeElement;
+      return of({ scrollTop, scrollHeight, clientHeight });
```
Comment (nnixaa) [inline src/framework/theme/components/list/infinite-list.directive.ts]: of -> observableOf

File: src/playground/infinite-list/infinite-news-list.component.ts [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    private newsService: NewsService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private scrollService: NbLayoutScrollService,
+    @Inject(PLATFORM_ID) private platformId,
+    @Inject(NB_WINDOW) private window,
+  ) {
+    if (isPlatformBrowser(this.platformId) && this.window.history.scrollRestoration) {
+      this.initialScrollRestoration = window.history.scrollRestoration;
+      history.scrollRestoration = 'manual';
```
Comment (nnixaa) [inline src/playground/infinite-list/infinite-news-list.component.ts]: we should add a not about this

File: src/playground/infinite-list/infinite-news-list.component.ts [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+  }
+
+  ngOnDestroy() {
+    if (this.initialScrollRestoration) {
+      this.window.history.scrollRestoration = this.initialScrollRestoration;
+    }
+  }
+
+  updateUrl(page) {
+    const queryParams = { ...this.route.snapshot.queryParams, page };
```
Comment (nnixaa) [inline src/playground/infinite-list/infinite-news-list.component.ts]: we might not need to merge it here as `navigate` can do this internally https://angular.io/api/router/NavigationExtras#queryParamsHandling

File: src/playground/infinite-list/news-post.component.ts [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+    <div class="text-placeholder"></div>
+    <div class="link-placeholder"></div>
+  `,
+  styles: [`
+    :host {
+      display: block;
+    }
+
+    [class$="placeholder"] {
+      background: #f1f2f3;
```
Comment (nnixaa) [inline src/playground/infinite-list/news-post.component.ts]: what about `cosmic`?

**Comment by nnixaa** [comment]
@yggg 🎉

**Pull request merged**